src/environment.cpp

```diff
@@ -589,34 +589,20 @@ class ABMHandler
 			if(aabm.chance == 0)
 				aabm.chance = 1;
 			// Trigger neighbors
-			std::set<std::string> required_neighbors_s
-					= abm->getRequiredNeighbors();
-			for(std::set<std::string>::iterator
-					i = required_neighbors_s.begin();
-					i != required_neighbors_s.end(); ++i)
-			{
-				ndef->getIds(*i, aabm.required_neighbors);
-			}
+			aabm.required_neighbors	= abm->getRequiredNeighbors();
 			// Trigger contents
-			std::set<std::string> contents_s = abm->getTriggerContents();
-			for(std::set<std::string>::iterator
-					i = contents_s.begin(); i != contents_s.end(); ++i)
-			{
-				std::set<content_t> ids;
-				ndef->getIds(*i, ids);
-				for(std::set<content_t>::const_iterator k = ids.begin();
-						k != ids.end(); ++k)
-				{
-					content_t c = *k;
-					std::map<content_t, std::vector<ActiveABM> >::iterator j;
+			std::vector<content_t> ids = abm->getTriggerContents();
+			for(std::vector<content_t>::const_iterator k = ids.begin();
```

Comment:
Missing a space between for and parentheses.
Vectors should use array notation instead of iterators.